osdep/compiler.h

```diff
@@ -3,7 +3,7 @@
 
 #define MP_EXPAND_ARGS(...) __VA_ARGS__
 
-#ifdef __GNUC__
+#if defined(__GNUC__) || defined(__clang__)
```

Comment:
Yes. 

Just to clear naming. Not sure what you call `MSVC mode`, but neither clang-cl (msvc driver) nor clang (gnu driver) defines `__GNUC__`. But both support clang's fronted features when targeting `-pc-windows-msvc`.

Reply:
Naming thing. `clang` mode for me is driver mode, GUN or MSVC, mostly meaning that `clang` support GNU-like CLI args and `clang-cl` supports MSVC-link CLI args.

Both of them can support different platform targets. Which can be `x86_64-w64-windows-gnu` or `x86_64-pc-windows-msvc` and after this PR both will be supported. Sorry for confusion, because this is what you call "mode" probably.

I should have time tonight to finish remaining bits, it will be clearer with full pipeline integrated.